### PR TITLE
July infra rollout: Move runtime tests out of the coreclr folder (2nd attempt)

### LINKED ISSUE #37440: July infra rollout: move tests shared between coreclr and mono from src/coreclr/tests/src to src/tests
As what was previously the "CoreCLR test suite" is now shared with mono, it no longer makes sense for it to sit under src/coreclr/tests in the runtime tree. The purpose of this task is to move the tests into a new common folder under the tree and modify the paths in all scripts and pipelines so that they continue running after the change. This task doesn't address any changes to the underlying harness used to run the tests.

Current root location of CoreCLR test source code and scripts:

<code>src/coreclr/tests/src</code>

Proposed new location for the test root:

<code>src/tests</code>

/cc: @dotnet/runtime-infrastructure

### PR DESCRIPTION
**Pull request by trylek**
I have messed up my previous version of this change as of https://github.com/dotnet/runtime/pull/37995 while rebasing the change against current master. The current expectation is that the change is going to be merged in around 7/6 i.e. at the July infra rollout, fixing the bug https://github.com/dotnet/runtime/issues/37440.

Thanks

Tomas

Fixes: https://github.com/dotnet/runtime/issues/37440

/cc: @dotnet/runtime-infrastructure

### CODE REVIEW

**Comment by safern** [comment]
@trylek does this include the runtimetests subset update in `runtime.yml` + running the coreclr tests when that changed?

**Comment by trylek** [comment]
@safern - not yet, working on that right now to make this a third PR feedback commit.

**Comment by trylek** [comment]
@safern - hopefully fixed, can you please take another look when you have a chance?

File: eng/pipelines/runtime.yml [hunk 1]
```diff
@@ -109,7 +109,9 @@ jobs:
       - eng/pipelines/installer/*
     - subset: runtimetests
       include:
+      - src/tests/*
       - src/coreclr/tests/*
```
Comment (safern) [inline eng/pipelines/runtime.yml:113]: Will we still have `src/coreclr/tests/*` be part of the runtime tests?

File: eng/pipelines/runtime.yml [hunk 2]
```diff
@@ -109,7 +109,9 @@ jobs:
...
+      - src/coreclr/build-test.cmd
```
Comment (safern) [inline eng/pipelines/runtime.yml:114]: aren't we going to move this scripts to the new folder?

**Comment by safern** [comment]
You'll also need to update the conditions to some jobs so that they run when the runtime tests are changed:

https://github.com/dotnet/runtime/blob/db91329529e31a10f176e477de0a71b935d4575a/eng/pipelines/runtime.yml#L155
https://github.com/dotnet/runtime/blob/db91329529e31a10f176e477de0a71b935d4575a/eng/pipelines/runtime.yml#L190
https://github.com/dotnet/runtime/blob/db91329529e31a10f176e477de0a71b935d4575a/eng/pipelines/runtime.yml#L575
https://github.com/dotnet/runtime/blob/db91329529e31a10f176e477de0a71b935d4575a/eng/pipelines/runtime.yml#L613
https://github.com/dotnet/runtime/blob/db91329529e31a10f176e477de0a71b935d4575a/eng/pipelines/runtime.yml#L636
https://github.com/dotnet/runtime/blob/db91329529e31a10f176e477de0a71b935d4575a/eng/pipelines/runtime.yml#L655

File: eng/pipelines/runtime.yml (hunk 1 above)
Comment (trylek) [reply eng/pipelines/runtime.yml:113]: As I described in the infra standup, my current goal was to make the change as small as possible as the bulk test move makes it hard to manipulate and maintain. As part of this I haven't yet moved everything out of the <code>src/coreclr/tests</code> folder as that would require additional changes to build-test and elsewhere. I plan to do this as a second step, after the tests have been physically moved, and then I'll delete this clause.

File: eng/pipelines/runtime.yml (hunk 2 above)
Comment (trylek) [reply eng/pipelines/runtime.yml:114]: Ultimately, yes, but that would again require additional changes to the pipelines. As mentioned above, I would prefer to tackle this as a second step once the bulk test move has been merged in.

File: eng/pipelines/runtime.yml (hunk 1 above)
Comment (safern) [reply eng/pipelines/runtime.yml:113]: Ok just checking, makes sense. Thanks for elaborating here.

File: eng/pipelines/runtime.yml (hunk 2 above)
Comment (safern) [reply eng/pipelines/runtime.yml:114]: Makes sense.

**Comment by trylek** [comment]
@dotnet/runtime-infrastructure - I believe I'm finally out of the woods w.r.t. the weird difference between PR and CI runs of the change due to an omission in build-test.sh (not passing unprocessed build args to msbuild). I have triggered a CoreCLR outerloop run and I intend to coordinate tomorrow European morning with @ViktorHofer on finally merging the change in unless I hear objections - I would love to avoid further delays if possible, I hit two conflicts just this afternoon during lab testing, the change is vast and the test churn makes it really painful.

Thanks a lot

Tomas

File: eng/pipelines/runtime.yml [hunk 3]
```diff
@@ -598,6 +602,7 @@ jobs:
           eq(dependencies.checkout.outputs['SetPathVars_libraries.containsChange'], true),
           eq(dependencies.checkout.outputs['SetPathVars_coreclr.containsChange'], true),
           eq(dependencies.checkout.outputs['SetPathVars_mono.containsChange'], true),
+          eq(dependencies.checkout.outputs['SetPathVars_runtimetests.containsChange'], true),
```
Comment (safern) [inline eng/pipelines/runtime.yml:605]: Just double checking to refresh my memory, libraries tests are needed for runtime tests, right?

**Review by safern — APPROVED** [review]
Skimmed through

File: eng/pipelines/runtime.yml (hunk 3 above)
Comment (trylek) [reply eng/pipelines/runtime.yml:605]: I don't think we need any library test build artifacts for building or running runtime tests, just the libraries build. I can easily remove this line if you believe it's superfluous, I guess I'll probably catch a few more conflicts before the night is over, I just hope that around PST midnight the churn will calm down a bit. I'm not going to update the PR right away as that would cancel the running tests and that would be a waste of lab resources considering this is ultimately just a "perf optimization".

**Comment by trylek** [comment]
One more comment - I'm planning to merge this in without squashing i.e. as two commits, one doing the mechanical move and another with the script patches, for easier orientation. Please let me know if for whatever reason you find a full squash merge preferable.

File: eng/pipelines/runtime.yml (hunk 3 above)
Comment (safern) [reply eng/pipelines/runtime.yml:605]: I think we can update this as a subsequent PR to not risk getting more merge conflicts.

**Review by ViktorHofer — APPROVED** [review]

**Pull request merged**